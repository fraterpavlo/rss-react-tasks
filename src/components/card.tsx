import React from 'react';

export interface ICarData {
  id: string;
  brand: string;
  model: string;
  country: string;
  count: string;
  year: number;
  type: string;
  color: string;
  price: number;
  electric: boolean;
}

interface CardProps {
  cardData: ICarData;
}

export function Card({ cardData }: CardProps) {
  const isElectricItemStr: string = cardData.electric ? 'Да' : 'Нет';

  return (
    <div data-id={cardData.id} className="cards-area__item card">
      <span className="card__title">
        {cardData.brand} {cardData.model}
      </span>
      <div className="card__img-contain">
        <img src={`./assets/img/cars/${cardData.id}.jpg`} alt="car" className="card__img" />
      </div>
      <div className="card__type">
        Тип кузова:
        <span className="card__type-output">{cardData.type}</span>
      </div>
      <div className="card__country">
        Страна сборки:
        <span className="card__country-output">{cardData.country}</span>
      </div>
      <div className="card__year">
        Год производства:
        <span className="card__year-output">{cardData.year}</span>
      </div>
      <div className="card__price">
        Цена от:
        <span className="card__price-output">{cardData.price} &#128176;</span>
      </div>
      <div className="card__electric">
        Электрокар:
        <span className="card__electric-output">{isElectricItemStr}</span>
      </div>
    </div>
  );
}

// export default Card;
