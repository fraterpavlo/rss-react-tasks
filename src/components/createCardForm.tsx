// import React, { useRef, useState } from 'react';
// import MyLabel from './UI/myLabel';
// import MyInput from './UI/myInput';
// import MyButton from 'components/UI/myButton';
// import { CreateCardFormProps } from '../interfaces/components/createCardForm';
// import MySelect from './UI/mySelect';
// import classes from 'styles/components/create-card-form.module.css';

// const CreateCardForm = ({ create }: CreateCardFormProps) => {
//   const submitBtnRef = useRef<HTMLButtonElement | null>(null);
//   const carBrandInputRef = useRef<HTMLInputElement | null>(null);
//   const carModelInputRef = useRef<HTMLInputElement | null>(null);
//   const carTypeSelectRef = useRef<HTMLSelectElement | null>(null);
//   // const carColorFormRef = useRef<HTMLSelectElement | null>(null);
//   const carBuildCountrySelectRef = useRef<HTMLSelectElement | null>(null);
//   const carBuildDateInputRef = useRef<HTMLInputElement | null>(null);
//   const carPriceInputRef = useRef<HTMLInputElement | null>(null);
//   const carIsElectricInputRef = useRef<HTMLInputElement | null>(null);
//   const carFileInputRef = useRef<HTMLInputElement | null>(null);
//   const [messages, setMessages] = useState({
//     carBrandInputMessage: '',
//     carModelInputMessage: '',
//     carBuildDateInputMessage: '',
//     carPriceInputMessage: '',
//     carFileInputMessage: '',
//   });

//   function checkIsValidFormValues() {
//     const CarBrandInputValue = carBrandInputRef.current?.value;
//     const CarModelInputValue = carModelInputRef.current?.value;
//     const CarBuildDateInputValue = carBuildDateInputRef.current?.value;
//     const CarPriceInputValue = carPriceInputRef.current?.value;
//     const CarFileInputFiles = carFileInputRef.current!.files;
//     let isValidCarBrandInput = false;
//     let isValidCarModelInput = false;
//     let isValidСarBuildDateInput = false;
//     let isValidCarPriceInput = false;
//     let isValidCarFileInput = false;

//     if (CarBrandInputValue?.trim().length === 0) {
//       setMessages({
//         ...messages,
//         carBrandInputMessage: 'Введите марку автомобиля',
//       });
//     } else if (/[0-9]/.test(CarBrandInputValue!)) {
//       setMessages({
//         ...messages,
//         carBrandInputMessage: 'Марка автомобиля не должна содержать цифр',
//       });
//     } else if (CarBrandInputValue!.length > 50) {
//       setMessages({
//         ...messages,
//         carBrandInputMessage: 'Марка автомобиля должна содержать менее 50 символов',
//       });
//     } else {
//       isValidCarBrandInput = true;
//       setMessages({
//         ...messages,
//         carBrandInputMessage: '',
//       });
//     }

//     if (CarModelInputValue?.trim().length === 0) {
//       setMessages({
//         ...messages,
//         carModelInputMessage: 'Введите модель автомобиля',
//       });
//     } else if (CarBrandInputValue!.length > 50) {
//       setMessages({
//         ...messages,
//         carModelInputMessage: 'Модель автомобиля должна содержать менее 50 символов',
//       });
//     } else {
//       isValidCarModelInput = true;
//       setMessages({
//         ...messages,
//         carModelInputMessage: '',
//       });
//     }

//     if (!Number(CarBuildDateInputValue!.split('-')[0])) {
//       setMessages({
//         ...messages,
//         carBuildDateInputMessage: 'Введите год производства автомобиля',
//       });
//     } else {
//       setMessages({
//         ...messages,
//         carBuildDateInputMessage: '',
//       });
//       isValidСarBuildDateInput = true;
//     }

//     if (!Number(CarPriceInputValue)) {
//       setMessages({
//         ...messages,
//         carPriceInputMessage: 'Введите цену автомобиля',
//       });
//     } else {
//       setMessages({
//         ...messages,
//         carPriceInputMessage: '',
//       });
//       isValidCarPriceInput = true;
//     }

//     if (!CarFileInputFiles![0]) {
//       setMessages({
//         ...messages,
//         carFileInputMessage: 'Выберите фото автомобиля',
//       });
//     } else {
//       setMessages({
//         ...messages,
//         carFileInputMessage: '',
//       });
//       isValidCarFileInput = true;
//     }

//     if (
//       isValidCarBrandInput &&
//       isValidCarModelInput &&
//       isValidСarBuildDateInput &&
//       isValidCarPriceInput &&
//       isValidCarFileInput
//     )
//       submitBtnRef.current!.disabled = false;
//   }

//   const addNewCard = (e: React.MouseEvent<HTMLButtonElement>) => {
//     e.preventDefault();
//     console.log(111111111111);

//     const newCardData = {
//       id: `${new Date()}`,
//       brand: `${carBrandInputRef.current?.value.trim()}`,
//       model: `${carModelInputRef.current?.value.trim()}`,
//       country: `${carBuildCountrySelectRef.current?.value}`,
//       count: `${Math.random() * 10}`,
//       year: Number(carBuildDateInputRef.current?.value.split('-')[0]),
//       type: `${carTypeSelectRef.current?.value}`,
//       color: 'unknown',
//       price: Number(carPriceInputRef.current?.value),
//       electric: carIsElectricInputRef.current!.checked,
//       imageSrc: carFileInputRef.current!.files![0]
//         ? URL.createObjectURL(carFileInputRef.current!.files![0])
//         : '',
//     };
//     create(newCardData);
//   };

//   return (
//     <form className={classes.createCardForm}>
//       <MyLabel tittle={'Марка Авто'} message={messages.carBrandInputMessage}>
//         <MyInput
//           defaultValue={''}
//           // value={''}
//           onChange={checkIsValidFormValues}
//           ref={carBrandInputRef}
//           type="text"
//           placeholder="Марка"
//         />
//       </MyLabel>
//       <MyLabel tittle={'Модель'} message={messages.carModelInputMessage}>
//         <MyInput
//           defaultValue={''}
//           ref={carModelInputRef}
//           onChange={checkIsValidFormValues}
//           type="text"
//           placeholder="Модель"
//         />
//       </MyLabel>
//       <MyLabel tittle={'Тип кузова'}>
//         <MySelect
//           ref={carTypeSelectRef}
//           value={''}
//           onChange={(selectedType) => console.log(selectedType)}
//           defaultValue="Тип кузова"
//           options={[
//             { value: 'купе', name: 'купе' },
//             { value: 'кабриолет', name: 'кабриолет' },
//             { value: 'минивэн', name: 'минивэн' },
//             { value: 'кроссовер', name: 'кроссовер' },
//             { value: 'хэтчбек', name: 'хэтчбек' },
//             { value: 'седан', name: 'седан' },
//             { value: 'родстер', name: 'родстер' },
//             { value: 'внедорожник', name: 'внедорожник' },
//             { value: 'пикап', name: 'пикап' },
//             { value: 'фургон', name: 'фургон' },
//           ]}
//         ></MySelect>
//       </MyLabel>
//       <MyLabel tittle={'Страна производства'}>
//         <MySelect
//           ref={carBuildCountrySelectRef}
//           value={''}
//           onChange={(selectedType) => console.log(selectedType)}
//           defaultValue="Страна производства"
//           options={[
//             { value: 'Великобритания', name: 'Великобритания' },
//             { value: 'Испания', name: 'Испания' },
//             { value: 'Германия', name: 'Германия' },
//             { value: 'Словакия', name: 'Словакия' },
//             { value: 'Бельгия', name: 'Бельгия' },
//             { value: 'Россия', name: 'Россия' },
//             { value: 'Австрия', name: 'Австрия' },
//             { value: 'Беларусь', name: 'Беларусь' },
//             { value: 'США', name: 'США' },
//             { value: 'Италия', name: 'Италия' },
//           ]}
//         ></MySelect>
//       </MyLabel>

//       <MyLabel tittle={'Дата производства'}>
//         <MyInput
//           defaultValue={''}
//           ref={carBuildDateInputRef}
//           onChange={checkIsValidFormValues}
//           type="date"
//           required
//         />
//       </MyLabel>

//       <MyLabel tittle={'Цена'} message={messages.carPriceInputMessage}>
//         <MyInput
//           defaultValue={''}
//           ref={carPriceInputRef}
//           onChange={checkIsValidFormValues}
//           type="number"
//           required
//         />
//       </MyLabel>

//       <MyLabel tittle={'Электрокар'}>
//         <MyInput defaultValue={''} ref={carIsElectricInputRef} type="checkbox" required />
//       </MyLabel>

//       <MyLabel tittle={'загрузить файл'} message={messages.carFileInputMessage}>
//         <MyInput
//           defaultValue={''}
//           ref={carFileInputRef}
//           onChange={checkIsValidFormValues}
//           type="file"
//           accept="image/png, image/jpeg, image/jpeg"
//         />
//       </MyLabel>

//       <MyButton ref={submitBtnRef} onClick={addNewCard} disabled>
//         Создать карточку
//       </MyButton>
//     </form>
//   );
// };

// export default CreateCardForm;
