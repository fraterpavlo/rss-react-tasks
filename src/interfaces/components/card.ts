export interface ICarData {
  id: string;
  brand: string;
  model: string;
  country: string;
  count: string;
  year: number;
  type: string;
  color: string;
  price: number;
  electric: boolean;
  imageSrc?: string;
}

export interface CardProps {
  cardData: ICarData;
}
