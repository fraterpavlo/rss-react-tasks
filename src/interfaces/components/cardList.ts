import { Character } from 'rickmortyapi/dist/interfaces';

export interface CardsListProps {
  rootClasses?: string;
  dataList: Character[];
}
