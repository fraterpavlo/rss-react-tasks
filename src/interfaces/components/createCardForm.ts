import { VoidCallbackType } from '../common';

export interface ICarData {
  id: string;
  brand: string;
  model: string;
  country: string;
  count: string;
  year: number;
  type: string;
  color: string;
  price: number;
  electric: boolean;
}

export interface CreateCardFormProps {
  create: VoidCallbackType<ICarData>;
}
