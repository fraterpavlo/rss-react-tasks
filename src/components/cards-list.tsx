import React from 'react';

import { ICarData, Card } from './card';

interface CardsListProps {
  dataList: ICarData[];
}

function CardsList({ dataList }: CardsListProps) {
  const list = dataList.map((itemData: ICarData) => <Card key={itemData.id} cardData={itemData} />);
  return <div className="homepage__cards-area cards-area__container">{list}</div>;
}

export default CardsList;
