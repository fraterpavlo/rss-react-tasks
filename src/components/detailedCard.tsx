import React from 'react';
import { IDetailedCardProps } from '../interfaces/components/detailedCard';
import styles from 'styles/components/detailedCard.module.css';

function DetailedCard({ cardData }: IDetailedCardProps) {
  if (!cardData) return <strong>sorry, data not found</strong>;

  return (
    <div data-testid={cardData.id} data-card-id={cardData.id} className={styles.card}>
      <span className={styles['card__name']}>{cardData.name}</span>
      <div className={styles['card__img-contain']}>
        <img src={cardData.image} alt="card image" className={styles['card__img']} />
      </div>
      <div className="card__gender">
        gender:
        <span className="card__gender_output"> {cardData.gender}</span>
      </div>
      <div className="card__species">
        species:
        <span className="card__species_output"> {cardData.species}</span>
      </div>
      <div className="card__status">
        status:
        <span className="card__status_output"> {cardData.status}</span>
      </div>
      <div className="card__origin">
        origin:
        <span className="card__origin_output"> {cardData.origin.name}</span>
      </div>
      <div className="card__location">
        location:
        <span className="card__location_output"> {cardData.location.name}</span>
      </div>
      <div className="card__created">
        created:
        <span className="card__created_output">
          {` ${new Date(cardData.created).getDay()}-
          ${new Date(cardData.created).getMonth()}-
          ${new Date(cardData.created).getFullYear()}`}
        </span>
      </div>
    </div>
  );
}

export default DetailedCard;
