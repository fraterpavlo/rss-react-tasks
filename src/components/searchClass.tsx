import React from 'react';
import TextField from '@mui/material/TextField';
import { SearchClassProps } from '../interfaces/components/searchClass';

export default class extends React.Component<SearchClassProps> {
  state = {
    currentValue: localStorage.getItem('search-value') || '',
  };

  componentWillUnmount() {
    localStorage.setItem('search-value', this.state.currentValue);
  }

  onSearchChangeHandler(value: string) {
    this.setState({ currentValue: value });
    this.props.passingSearchStateToParent(value);
  }

  render() {
    return (
      <TextField
        className="homepage__textfield"
        id="outlined-search"
        label="Search field"
        type="search"
        autoComplete="off"
        value={this.state.currentValue}
        onChange={(e) => {
          this.onSearchChangeHandler(e.target.value);
        }}
      />
    );
  }
}
