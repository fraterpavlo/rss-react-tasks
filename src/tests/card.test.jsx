import React from 'react';
import { render } from '@testing-library/react';
// import CardsList from 'components/cards-list';
// import carsDataList from 'data';
import { Card } from 'components/card';

const testCarData = {
  id: 'test',
  brand: 'test',
  model: 'test',
  country: 'test',
  count: 'test',
  year: 9999,
  type: 'test',
  color: 'test',
  price: 999,
  electric: false,
};

describe('test card', () => {
  it('tets card match snapshot', () => {
    const card = render(<Card cardData={testCarData} />);
    expect(card).toMatchSnapshot();
  });
});
