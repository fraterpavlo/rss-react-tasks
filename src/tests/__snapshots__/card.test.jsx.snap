// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test card tets card match snapshot 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="cards-area__item card"
        data-id="test"
      >
        <span
          class="card__title"
        >
          test
           
          test
        </span>
        <div
          class="card__img-contain"
        >
          <img
            alt="car"
            class="card__img"
            src="./assets/img/cars/test.jpg"
          />
        </div>
        <div
          class="card__type"
        >
          –¢–∏–ø –∫—É–∑–æ–≤–∞:
          <span
            class="card__type-output"
          >
            test
          </span>
        </div>
        <div
          class="card__country"
        >
          –°—Ç—Ä–∞–Ω–∞ —Å–±–æ—Ä–∫–∏:
          <span
            class="card__country-output"
          >
            test
          </span>
        </div>
        <div
          class="card__year"
        >
          –ì–æ–¥ –ø—Ä–æ–∏–∑–≤–æ–¥—Å—Ç–≤–∞:
          <span
            class="card__year-output"
          >
            9999
          </span>
        </div>
        <div
          class="card__price"
        >
          –¶–µ–Ω–∞ –æ—Ç:
          <span
            class="card__price-output"
          >
            999
             üí∞
          </span>
        </div>
        <div
          class="card__electric"
        >
          –≠–ª–µ–∫—Ç—Ä–æ–∫–∞—Ä:
          <span
            class="card__electric-output"
          >
            –ù–µ—Ç
          </span>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="cards-area__item card"
      data-id="test"
    >
      <span
        class="card__title"
      >
        test
         
        test
      </span>
      <div
        class="card__img-contain"
      >
        <img
          alt="car"
          class="card__img"
          src="./assets/img/cars/test.jpg"
        />
      </div>
      <div
        class="card__type"
      >
        –¢–∏–ø –∫—É–∑–æ–≤–∞:
        <span
          class="card__type-output"
        >
          test
        </span>
      </div>
      <div
        class="card__country"
      >
        –°—Ç—Ä–∞–Ω–∞ —Å–±–æ—Ä–∫–∏:
        <span
          class="card__country-output"
        >
          test
        </span>
      </div>
      <div
        class="card__year"
      >
        –ì–æ–¥ –ø—Ä–æ–∏–∑–≤–æ–¥—Å—Ç–≤–∞:
        <span
          class="card__year-output"
        >
          9999
        </span>
      </div>
      <div
        class="card__price"
      >
        –¶–µ–Ω–∞ –æ—Ç:
        <span
          class="card__price-output"
        >
          999
           üí∞
        </span>
      </div>
      <div
        class="card__electric"
      >
        –≠–ª–µ–∫—Ç—Ä–æ–∫–∞—Ä:
        <span
          class="card__electric-output"
        >
          –ù–µ—Ç
        </span>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
